datafusion/core/src/physical_plan/windows/mod.rs

```diff
@@ -90,12 +93,26 @@ pub fn create_window_expr(
             order_by,
             window_frame,
         )),
-        WindowFunction::AggregateUDF(fun) => Arc::new(PlainAggregateWindowExpr::new(
-            udaf::create_aggregate_expr(fun.as_ref(), args, input_schema, name)?,
-            partition_by,
-            order_by,
-            window_frame,
-        )),
+        WindowFunction::AggregateUDF(fun) => {
+            let aggregate =
+                udaf::create_aggregate_expr(fun.as_ref(), args, input_schema, name)?;
+
+            if !unbounded_window && aggregate.retractable()? {
```

Comment:
I think this line should be
```suggestion
            if !unbounded_window  {
```
If aggregate doesn't implement retract we should get an error. There is no way to run aggregator whose start is different than (`UNBOUNDED PRECEDING`) without `retract_batch` method.

Reply:
I will make this change -- thank you @mustafasrepo  -- I didn't appreciate that there is no correct way to run the AggregateUDF correctly without retract_batch. Thus I think the correct solution is to return an error when such a combination is attempted. I will update the PR to reflect this

Reply:
in https://github.com/apache/arrow-datafusion/pull/6671/commits/999866e281e454ff8824829fce36ee026140e143